isofit/core/forward.py

```diff
@@ -104,6 +106,19 @@ def __init__(self, full_config: Config):
             raise ValueError("Must specify a valid surface model")
             # No need to be more specific - should have been checked in config already
 
+        # check if wavelength grid provided by surface model matches instrument wavelengths
+        if self.surface.n_wl != len(self.RT.wl):
+            raise ValueError(
+                "Number of channels provided in surface model file does not match"
+                " wavelengths in radiance cube. Please rebuild your surface model."
+            )
+        if not all(self.surface.wl == self.RT.wl):
+            Logger.warning(
+                "Center wavelengths provided in surface model file do not match"
+                " wavelengths in radiance cube. Please consider rebuilding your"
+                " surface model for optimal performance."
+            )
```

Comment:
Is there an easy way to fold this into the configs?  Perhaps not, if data hasn't been loaded.  But anything we can push in that direction makes life easier for users, as the error happens sooner.

Reply:
That's indeed difficult to achieve. Right now, we only have access to the wavelength grid of the surface model after initializing the respective surface class within the `init` of the forward model class. We could move the check to `apply_oe.py` itself by loading the surface file there and comparing its wavelengths against the wavelength file and/or the grid from the radiance cube. That would actually be much better since we'd avoid running a lot of operations (obtain metadata, segmentation, etc.) before raising a potential exception. What do you think?

Reply:
Hmm, maybe let's do both?  I like this for the absence of apply_oe, but we can certainly use a check there as well.

Reply:
All right, that sounds good. I'll implement a similar check in `apply_oe.py` as well.

Reply:
See [3304b06](https://github.com/isofit/isofit/pull/476/commits/3304b06578c1f74749b10c226f0340b77c9fde6a).